server/src/main/java/org/opensearch/common/util/CollectionUtils.java

```diff
@@ -95,61 +93,6 @@ public static <T> List<T> rotate(final List<T> list, int distance) {
         return new RotatedList<>(list, d);
     }
 
-    public static void sortAndDedup(final ObjectArrayList<byte[]> array) {
-        int len = array.size();
-        if (len > 1) {
-            sort(array);
-            int uniqueCount = 1;
-            for (int i = 1; i < len; ++i) {
-                if (!Arrays.equals(array.get(i), array.get(i - 1))) {
-                    array.set(uniqueCount++, array.get(i));
-                }
-            }
-            array.elementsCount = uniqueCount;
-        }
-    }
-
-    public static void sort(final ObjectArrayList<byte[]> array) {
-        new IntroSorter() {
```

Comment:
@nknize the Lucene's sort implementation seems to be adaptive to the length of the collection to be sorted (quicksort -> heapsort). I don't know how important this is, do we want to at least prepare a few benchmarks to see the heap/ops tradeoffs between these two implementations? (particularly `TreeSet` is super convenient but it also uses more heap space comparing to simple list).